File: Byte High Frequency 50 Questions/20210911/reverseList.go
package _0210911

type ListNode struct {
	Val  int
	Next *ListNode
}

// ReverseList 定义一个函数，输入一个链表的头节点，反转该链表并输出反转后链表的头节点。
func ReverseList(head *ListNode) *ListNode {
	var pre *ListNode
	for head != nil {
		temp := head.Next
		head.Next = pre
		pre = head
		head = temp
	}
	return pre
}

File: 剑指offer/easy/20210913/reverseList.cpp
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        ListNode *pre=nullptr;
        ListNode *cur=head;
        while(cur!= nullptr){
            ListNode *temp=cur->next;
            cur->next=pre;
            pre=cur;
            cur=temp;
        }
        return pre;
    }
};
